addons/website_sale/tests/test_website_sale_visitor.py

```diff
@@ -69,16 +69,19 @@ def test_dynamic_filter_newest_products(self):
         self.website = self.website.with_user(public_user).with_context(website_id=self.website.id)
         snippet_filter = self.env.ref('website_sale.dynamic_filter_newest_products')
 
-        res = snippet_filter._prepare_values(16, [])
-        res_products = [res_product['_record'] for res_product in res]
-        self.assertIn(product, res_products)
+        so = self.env['sale.order'].create({'partner_id': public_user.id})
 
-        product.product_tmpl_id.company_id = new_company
-        product.product_tmpl_id.flush_recordset(['company_id'])
+        with MockRequest(self.env, website=self.website, sale_order_id=so.id) as dummy:
+            res = snippet_filter._prepare_values(16, [])
+            res_products = [res_product['_record'] for res_product in res]
+            self.assertIn(product, res_products)
 
-        res = snippet_filter._prepare_values(16, [])
-        res_products = [res_product['_record'] for res_product in res]
-        self.assertNotIn(product, res_products)
+            product.product_tmpl_id.company_id = new_company
+            product.product_tmpl_id.flush_recordset(['company_id'])
+
+            res = snippet_filter._prepare_values(16, [])
+            res_products = [res_product['_record'] for res_product in res]
+            self.assertNotIn(product, res_products)
```

Comment:
Why is this change actually necessary? does `_prepare_values` now depend on `request`?